### ISSUE TITLE
Bug: [consistent-type-imports] "used by decorator metadata" false positive for type used in constructor parameter

### ISSUE THREAD

**Issue by mpotthoff** [typescript-eslint/typescript-eslint#7327]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

When upgrading from typescript-eslint 5.x.x to 6.x.x I started seeing false positive errors "used by decorator metadata" for types that are used in constructor parameters, even if the class itself is not using any decorators at all.

So given these two files
```typescript
export class ClassA {}
```

```typescript
import type { ClassA } from "./classA";

export class ClassB {
    public constructor(classA: ClassA) {}
}
```

we would expect no errors, as no decorators are involved in either file.

Instead we get an error reported for the import of `ClassA`:
```
1:1  error  Type import "ClassA" is used by decorator metadata  @typescript-eslint/consistent-type-imports
```

### Reproduction Repository Link

https://github.com/mpotthoff/consistent-type-imports-decorator-bug

### Repro Steps

1) clone the repo
2) `npm install`
3) `npx eslint classB.ts`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.2.0` |
| `@typescript-eslint/parser`            | `6.2.0` |
| `@typescript-eslint/scope-manager`     | `6.2.0` |
| `@typescript-eslint/typescript-estree` | `6.2.0` |
| `@typescript-eslint/type-utils`        | `6.2.0` |
| `@typescript-eslint/utils`             | `6.2.0` |
| `TypeScript`                           | `5.1.6` |
| `ESLint`                               | `8.45.0` |
| `node`                                 | `18.16.1` |

**Comment by bradzacher** [comment]
bug is this line here:
https://github.com/typescript-eslint/typescript-eslint/blob/a56af12b83fab9def8989a83d5b9d2a660b099fd/packages/scope-manager/src/referencer/ClassVisitor.ts#L195

It was missed in #6274 (cc @JoshuaKGoldberg we need to turn on `no-unnecessary-condition`)

**Comment by JoshuaKGoldberg** [comment]
Oop. I fudged it, I'll fix it.
